src/Metadata/Resource/Factory/CachedResourceNameCollectionFactory.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ApiPlatform\Core\Metadata\Resource\Factory;
+
+use ApiPlatform\Core\Metadata\Resource\ResourceNameCollection;
+use Psr\Cache\CacheException;
+use Psr\Cache\CacheItemPoolInterface;
+
+/**
+ * Caches resource name collection.
+ *
+ * @author Teoh Han Hui <teohhanhui@gmail.com>
+ */
+final class CachedResourceNameCollectionFactory implements ResourceNameCollectionFactoryInterface
+{
+    const CACHE_KEY_PREFIX = 'resource_name_collection_';
+
+    private $cacheItemPool;
+    private $decorated;
+
+    public function __construct(CacheItemPoolInterface $cacheItemPool, ResourceNameCollectionFactoryInterface $decorated)
+    {
+        $this->cacheItemPool = $cacheItemPool;
+        $this->decorated = $decorated;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function create() : ResourceNameCollection
+    {
+        $cacheKey = self::CACHE_KEY_PREFIX.md5(json_encode([]));
```

Comment:
This is correct. There are no parameters here. I'm only using a placeholder, which is of course unnecessary but I don't see why not (performance difference is negligible).